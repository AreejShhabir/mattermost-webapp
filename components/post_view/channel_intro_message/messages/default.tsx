// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';

import {FormattedMessage} from 'react-intl';

import AddGroupsToTeamModal from 'components/add_groups_to_team_modal';

import {Permissions} from 'mattermost-redux/constants';

import TeamPermissionGate from 'components/permissions_gates/team_permission_gate';
import {t} from 'utils/i18n';
import {Constants, ModalIdentifiers} from 'utils/constants';
import ToggleModalButton from 'components/toggle_modal_button';

import LocalizedIcon from 'components/localized_icon';
import {Channel} from '@mattermost/types/channels';
import ChannelPermissionGate from 'components/permissions_gates/channel_permission_gate';
import {PluginComponent} from 'types/store/plugins';
import FormattedMarkdownMessage from 'components/formatted_markdown_message';
import AddMembersButton from '../add_members_button';
import BoardsButton from '../boards_button';
import SetHeaderButton from '../set_header_button';

type Props = {
    channel: Channel;
    stats: any;
    usersLimit: number;
    boardComponent?: PluginComponent;
    enableUserCreation?: boolean;
    isReadOnly?: boolean;
    teamIsGroupConstrained?: boolean;
}

const DefaultIntroMessage = ({
    channel,
    stats,
    usersLimit,
    boardComponent,
    enableUserCreation,
    isReadOnly,
    teamIsGroupConstrained,
}: Props) => {
    let teamInviteLink = null;
    const totalUsers = stats.total_users_count;
    const isPrivate = channel.type === Constants.PRIVATE_CHANNEL;

    let setHeaderButton = null;
    let boardCreateButton = null;
    if (!isReadOnly) {
        boardCreateButton = (
            <BoardsButton
                channel={channel}
                boardComponent={boardComponent}
            />
        );
        const children = (
            <SetHeaderButton
                channel={channel}
            />
        );
        if (children) {
            setHeaderButton = (
                <ChannelPermissionGate
                    teamId={channel.team_id}
                    channelId={channel.id}
                    permissions={[isPrivate ? Permissions.MANAGE_PRIVATE_CHANNEL_PROPERTIES : Permissions.MANAGE_PUBLIC_CHANNEL_PROPERTIES]}
                >
                    {children}
                </ChannelPermissionGate>
            );
        }
    }

    if (!isReadOnly && enableUserCreation) {
        teamInviteLink = (
            <TeamPermissionGate
                teamId={channel.team_id}
                permissions={[Permissions.INVITE_USER]}
            >
                <TeamPermissionGate
                    teamId={channel.team_id}
                    permissions={[Permissions.ADD_USER_TO_TEAM]}
                >
                    {!teamIsGroupConstrained &&
                        <AddMembersButton
                            setHeader={setHeaderButton}
                            totalUsers={totalUsers}
                            usersLimit={usersLimit}
                            channel={channel}
                            createBoard={boardCreateButton}
                        />
                    }
                    {teamIsGroupConstrained &&
                    <ToggleModalButton
                        className='intro-links color--link'
                        modalId={ModalIdentifiers.ADD_GROUPS_TO_TEAM}
                        dialogType={AddGroupsToTeamModal}
                        dialogProps={{channel}}
                    >
                        {/* MM-46602: convert to compass icon after localization is added */}
                        <LocalizedIcon
                            className='fa fa-user-plus'
                            title={{id: t('generic_icons.add'), defaultMessage: 'Add Icon'}}
                        />
                        <FormattedMessage
                            id='intro_messages.addGroupsToTeam'
                            defaultMessage='Add other groups to this team'
                        />
                    </ToggleModalButton>
                    }
                </TeamPermissionGate>
            </TeamPermissionGate>
        );
    }

    return (
        <>
            <h2 className='channel-intro__title'>
                <FormattedMessage
                    id='intro_messages.beginning'
                    defaultMessage='Beginning of {name}'
                    values={{
                        name: channel.display_name,
                    }}
                />
            </h2>
            <p className='channel-intro__content'>
                {!isReadOnly &&
                    <FormattedMarkdownMessage
                        id='intro_messages.default'
                        defaultMessage='**Welcome to {display_name}!**\n \nPost messages here that you want everyone to see. Everyone automatically becomes a permanent member of this channel when they join the team.'
                        values={{
                            display_name: channel.display_name,
                        }}
                    />
                }
                {isReadOnly &&
                    <FormattedMarkdownMessage
                        id='intro_messages.readonly.default'
                        defaultMessage='**Welcome to {display_name}!**\n \nMessages can only be posted by system admins. Everyone automatically becomes a permanent member of this channel when they join the team.'
                        values={{
                            display_name: channel.display_name,
                        }}
                    />
                }
            </p>
            {teamInviteLink}
            {teamIsGroupConstrained && boardCreateButton}
            {teamIsGroupConstrained && setHeaderButton}
            <br/>
        </>
    );
};

export default React.memo(DefaultIntroMessage);
