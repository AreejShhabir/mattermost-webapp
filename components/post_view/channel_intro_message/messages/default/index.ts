// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import {connect} from 'react-redux';

import {getConfig} from 'mattermost-redux/selectors/entities/general';
import {getCurrentChannel} from 'mattermost-redux/selectors/entities/channels';
import {getCurrentTeam} from 'mattermost-redux/selectors/entities/teams';
import {getTotalUsersStats as getTotalUsersStatsSelector} from 'mattermost-redux/selectors/entities/users';
import {GlobalState} from 'types/store';

import DefaultIntroMessage from './default';

function mapStateToProps(state: GlobalState) {
    const config = getConfig(state);
    const enableUserCreation = config.EnableUserCreation === 'true';
    const isReadOnly = false;
    const team = getCurrentTeam(state);
    const channel = getCurrentChannel(state) || {};
    const stats = getTotalUsersStatsSelector(state) || {total_users_count: 0};

    return {
        channel,
        enableUserCreation,
        isReadOnly,
        teamIsGroupConstrained: Boolean(team.group_constrained),
        stats,
    };
}

export default connect(mapStateToProps)(DefaultIntroMessage);
