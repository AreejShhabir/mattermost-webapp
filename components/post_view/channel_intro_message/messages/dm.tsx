// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';

import {FormattedMessage} from 'react-intl';

import {Channel} from '@mattermost/types/channels';
import ProfilePicture from 'components/profile_picture';
import {UserProfile as UserProfileRedux} from '@mattermost/types/users';
import {PluginComponent} from 'types/store/plugins';
import FormattedMarkdownMessage from 'components/formatted_markdown_message';
import UserProfile from 'components/user_profile';
import * as Utils from 'utils/utils';
import BoardsButton from '../boards_button';
import SetHeaderButton from '../set_header_button';

type Props = {
    boardComponent?: PluginComponent;
    centeredIntro: string;
    channel: Channel;
    teammate?: UserProfileRedux;
    teammateName?: string;
}

const DMIntroMessage = ({
    boardComponent,
    centeredIntro,
    channel,
    teammate,
    teammateName,
}: Props) => {
    const channelIntroId = 'channelIntro';
    if (teammate) {
        const src = teammate ? Utils.imageURLForUser(teammate.id, teammate.last_picture_update) : '';

        let setHeaderButton = null;
        let boardCreateButton = null;
        if (!teammate?.is_bot) {
            boardCreateButton = (
                <BoardsButton
                    channel={channel}
                    boardComponent={boardComponent}
                />
            );
            setHeaderButton = (
                <SetHeaderButton
                    channel={channel}
                />
            );
        }

        return (
            <div
                id={channelIntroId}
                className={'channel-intro ' + centeredIntro}
            >
                <div className='post-profile-img__container channel-intro-img'>
                    <ProfilePicture
                        src={src}
                        size='xl'
                        userId={teammate?.id}
                        username={teammate?.username}
                        hasMention={true}
                    />
                </div>
                <div className='channel-intro-profile d-flex'>
                    <UserProfile
                        userId={teammate?.id}
                        disablePopover={false}
                        hasMention={true}
                    />
                </div>
                <p className='channel-intro-text'>
                    <FormattedMarkdownMessage
                        id='intro_messages.DM'
                        defaultMessage='This is the start of your direct message history with {teammate}.\nDirect messages and files shared here are not shown to people outside this area.'
                        values={{
                            teammate: teammateName,
                        }}
                    />
                </p>
                {boardCreateButton}
                {setHeaderButton}
            </div>
        );
    }

    return (
        <div
            id={channelIntroId}
            className={'channel-intro ' + centeredIntro}
        >
            <p className='channel-intro-text'>
                <FormattedMessage
                    id='intro_messages.teammate'
                    defaultMessage='This is the start of your direct message history with this teammate. Direct messages and files shared here are not shown to people outside this area.'
                />
            </p>
        </div>
    );
};

export default React.memo(DMIntroMessage);
