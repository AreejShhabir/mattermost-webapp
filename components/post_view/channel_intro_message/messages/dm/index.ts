// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import {connect} from 'react-redux';

import {getCurrentChannel, getDirectTeammate} from 'mattermost-redux/selectors/entities/channels';
import {getChannelIntroPluginComponents} from 'selectors/plugins';
import {getDisplayNameByUser} from 'utils/utils';
import {GlobalState} from 'types/store';

import DMIntroMessage from './dm';

function mapStateToProps(state: GlobalState) {
    const channel = getCurrentChannel(state) || {};
    const teammate = getDirectTeammate(state, channel.id);
    const boardComponent = getChannelIntroPluginComponents(state).find((c) => c.pluginId === 'focalboard');

    return {
        channel,
        teammate,
        teammateName: getDisplayNameByUser(state, teammate),
        boardComponent,
    };
}

export default connect(mapStateToProps)(DMIntroMessage);
