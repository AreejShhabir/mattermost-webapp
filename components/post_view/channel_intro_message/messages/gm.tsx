// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';

import {FormattedMessage} from 'react-intl';

import {Channel} from '@mattermost/types/channels';
import ProfilePicture from 'components/profile_picture';
import {UserProfile as UserProfileRedux} from '@mattermost/types/users';
import {PluginComponent} from 'types/store/plugins';
import FormattedMarkdownMessage from 'components/formatted_markdown_message';
import * as Utils from 'utils/utils';
import BoardsButton from '../boards_button';
import SetHeaderButton from '../set_header_button';

type Props = {
    channel: Channel;
    currentUserId: string;
    profiles: UserProfileRedux[];
    boardComponent?: PluginComponent;
}

const GMIntroMessage = ({
    channel,
    currentUserId,
    profiles,
    boardComponent,
}: Props) => {
    const channelIsArchived = channel.delete_at !== 0;

    const renderButtons = !channelIsArchived;
    const boardCreateButton = renderButtons ? <BoardsButton boardComponent={boardComponent}/> : null;
    const setHeaderButton = renderButtons ? <SetHeaderButton channel={channel}/> : null;

    if (profiles.length > 0) {
        const pictures = profiles.
            filter((profile) => profile.id !== currentUserId).
            map((profile) => (
                <ProfilePicture
                    key={'introprofilepicture' + profile.id}
                    src={Utils.imageURLForUser(profile.id, profile.last_picture_update)}
                    size='xl'
                    userId={profile.id}
                    username={profile.username}
                />
            ));

        return (
            <>
                <div className='post-profile-img__container channel-intro-img'>
                    {pictures}
                </div>
                <p className='channel-intro-text'>
                    <FormattedMarkdownMessage
                        id='intro_messages.GM'
                        defaultMessage='This is the start of your group message history with {names}.\nMessages and files shared here are not shown to people outside this area.'
                        values={{
                            names: channel.display_name,
                        }}
                    />
                </p>
                {boardCreateButton}
                {setHeaderButton}
            </>
        );
    }

    return (
        <p className='channel-intro-text'>
            <FormattedMessage
                id='intro_messages.group_message'
                defaultMessage='This is the start of your group message history with these teammates. Messages and files shared here are not shown to people outside this area.'
            />
        </p>
    );
};

export default React.memo(GMIntroMessage);
