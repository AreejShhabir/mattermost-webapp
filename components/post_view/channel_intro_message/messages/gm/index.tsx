// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import {connect} from 'react-redux';

import {getCurrentChannel} from 'mattermost-redux/selectors/entities/channels';
import {getProfilesInCurrentChannel, getCurrentUserId} from 'mattermost-redux/selectors/entities/users';
import {GlobalState} from 'types/store';

import GMIntroMessage from './gm';

function mapStateToProps(state: GlobalState) {
    const channel = getCurrentChannel(state) || {};

    return {
        currentUserId: getCurrentUserId(state),
        channel,
        channelProfiles: getProfilesInCurrentChannel(state),
    };
}

export default connect(mapStateToProps)(GMIntroMessage);
