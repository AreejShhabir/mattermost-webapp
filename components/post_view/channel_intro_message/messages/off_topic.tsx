// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';

import {FormattedMessage} from 'react-intl';

import {Permissions} from 'mattermost-redux/constants';

import {Constants} from 'utils/constants';

import {Channel} from '@mattermost/types/channels';
import ChannelPermissionGate from 'components/permissions_gates/channel_permission_gate';
import {PluginComponent} from 'types/store/plugins';
import AddMembersButton from '../add_members_button';
import BoardsButton from '../boards_button';
import SetHeaderButton from '../set_header_button';

type Props = {
    channel: Channel;
    stats: any;
    usersLimit: number;
    boardComponent?: PluginComponent;
}

const OffTopicIntroMessage = ({
    channel,
    stats,
    usersLimit,
    boardComponent,
}: Props) => {
    const isPrivate = channel.type === Constants.PRIVATE_CHANNEL;
    const boardCreateButton = (
        <BoardsButton
            channel={channel}
            boardComponent={boardComponent}
        />
    );
    const children = (
        <SetHeaderButton
            channel={channel}
        />
    );
    const totalUsers = stats.total_users_count;

    let setHeaderButton = null;
    if (children) {
        setHeaderButton = (
            <ChannelPermissionGate
                teamId={channel.team_id}
                channelId={channel.id}
                permissions={[isPrivate ? Permissions.MANAGE_PRIVATE_CHANNEL_PROPERTIES : Permissions.MANAGE_PUBLIC_CHANNEL_PROPERTIES]}
            >
                {children}
            </ChannelPermissionGate>
        );
    }

    return (
        <>
            <h2 className='channel-intro__title'>
                <FormattedMessage
                    id='intro_messages.beginning'
                    defaultMessage='Beginning of {name}'
                    values={{
                        name: channel.display_name,
                    }}
                />
            </h2>
            <p className='channel-intro__content'>
                <FormattedMessage
                    id='intro_messages.offTopic'
                    defaultMessage='This is the start of {display_name}, a channel for non-work-related conversations.'
                    values={{
                        display_name: channel.display_name,
                    }}
                />
            </p>
            <AddMembersButton
                setHeader={setHeaderButton}
                totalUsers={totalUsers}
                usersLimit={usersLimit}
                channel={channel}
                createBoard={boardCreateButton}
            />
        </>
    );
};

export default React.memo(OffTopicIntroMessage);
