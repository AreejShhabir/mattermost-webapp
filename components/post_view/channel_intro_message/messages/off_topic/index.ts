// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import {connect} from 'react-redux';

import {getCurrentChannel} from 'mattermost-redux/selectors/entities/channels';
import {getTotalUsersStats as getTotalUsersStatsSelector} from 'mattermost-redux/selectors/entities/users';
import {GlobalState} from 'types/store';

import OffTopicIntroMessage from './off_topic';

function mapStateToProps(state: GlobalState) {
    const channel = getCurrentChannel(state) || {};
    const stats = getTotalUsersStatsSelector(state) || {total_users_count: 0};

    return {
        channel,
        stats,
    };
}

export default connect(mapStateToProps)(OffTopicIntroMessage);
