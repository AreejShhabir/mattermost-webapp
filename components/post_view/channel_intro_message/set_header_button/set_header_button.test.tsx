// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import {shallow} from 'enzyme';

import {archivedChannel, defaultChannel} from '../messages/test_utils';

import SetHeaderButton from './set_header_button';

const component = (otherProps: any) => (
    <SetHeaderButton
        channel={defaultChannel}
        {...otherProps}
    />
);

describe('components/post_view/ChannelIntroMessages', () => {
    describe('test Set Header Button', () => {
        test('should match snapshot', () => {
            expect(shallow(component({}))).toMatchSnapshot();
        });

        test('should match snapshot, has board, archived channel', () => {
            expect(shallow(component({
                channel: archivedChannel,
            }))).toMatchSnapshot();
        });
    });
});
