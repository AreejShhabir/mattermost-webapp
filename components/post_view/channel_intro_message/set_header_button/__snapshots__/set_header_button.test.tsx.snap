// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/post_view/ChannelIntroMessages test Set Header Button should match snapshot 1`] = `
<Connect(ChannelPermissionGate)
  channelId="channel_id"
  permissions={
    Array [
      "manage_public_channel_properties",
    ]
  }
  teamId="team-id"
>
  <Connect(ToggleModalButton)
    aria-label="Set a Header"
    className="intro-links color--link channelIntroButton"
    dialogProps={
      Object {
        "channel": Object {
          "create_at": 1508265709607,
          "creator_id": "creator_id",
          "delete_at": 0,
          "display_name": "test channel",
          "header": "test",
          "id": "channel_id",
          "last_post_at": 1508265709635,
          "name": "town-square",
          "purpose": "test",
          "team_id": "team-id",
          "type": "O",
          "update_at": 1508265709607,
        },
      }
    }
    dialogType={
      Object {
        "$$typeof": Symbol(react.memo),
        "WrappedComponent": [Function],
        "compare": null,
        "displayName": "Connect(EditChannelHeaderModal)",
        "type": [Function],
      }
    }
    modalId="edit_channel_header"
  >
    <EditIcon />
    Set a Header
  </Connect(ToggleModalButton)>
</Connect(ChannelPermissionGate)>
`;

exports[`components/post_view/ChannelIntroMessages test Set Header Button should match snapshot, has board, archived channel 1`] = `""`;

exports[`components/post_view/ChannelIntroMessages test Set Header Button should match snapshot, show false 1`] = `""`;
